conjure-java-core/src/test/java/com/palantir/conjure/java/types/WireFormatTests.java

```diff
@@ -154,9 +161,30 @@ public void testEmptyObjectsDeserialize() throws Exception {
     @Test
     public void testEnumCasingDeserializationInvariantToInputCase() throws Exception {
         assertThat(mapper.readValue("\"ONE\"", EnumExample.class)).isEqualTo(EnumExample.ONE);
-        assertThat(mapper.readValue("\"one\"", EnumExample.class)).isEqualTo(EnumExample.ONE);
-        assertThat(mapper.readValue("\"onE\"", EnumExample.class)).isEqualTo(EnumExample.ONE);
-        assertThat(mapper.readValue("\"oNE\"", EnumExample.class)).isEqualTo(EnumExample.ONE);
+        assertThat(mapper.readValue("\"ONE_HUNDRED\"", EnumExample.class)).isEqualTo(EnumExample.ONE_HUNDRED);
+        assertThatExceptionThrownRootCause(() -> mapper.readValue("\"one\"", EnumExample.class))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasLogMessage("Enum values must use SCREAMING_SNAKE_CASE")
+                .hasExactlyArgs(UnsafeArg.of("value", "one"));
+        assertThatExceptionThrownRootCause(() -> mapper.readValue("\"onE\"", EnumExample.class))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasLogMessage("Enum values must use SCREAMING_SNAKE_CASE")
+                .hasExactlyArgs(UnsafeArg.of("value", "onE"));
+        assertThatExceptionThrownRootCause(() -> mapper.readValue("\"oNE\"", EnumExample.class))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasLogMessage("Enum values must use SCREAMING_SNAKE_CASE")
```

Comment:
Could we tweak this error message to just say `UPPER_SNAKE_CASE`? I think the 'screaming' bit might be a bit more humour than java programmers can handle

Reply:
Sounds good, thanks Dan!